app/main.py

```diff
@@ -39,51 +46,274 @@ def execute(self):
+        dict_shelve = {}
+        if self.get_kytos_topology():
+            self.load_shelve()
+            # open the sdx topology shelve file
+            with shelve.open("topology_shelve") as open_shelve:
+                if 'id' in open_shelve.keys() and \
+                        'name' in open_shelve.keys() and \
+                        'version' in open_shelve.keys():
+                    # initialize version control
+                    if open_shelve['version'] == 0:
+                        open_shelve['version'] = 1
+                        self.version_control = True  # pylint: disable=W0201
+                    dict_shelve = dict(open_shelve)
+                    open_shelve.close()
+        return JSONResponse(dict_shelve)
+
+    # rest api tests
+    @rest("v1/validate_sdx_topology", methods=["POST"])
+    def get_validate_sdx_topology(self, request: Request) -> JSONResponse:
+        """ REST to return the validated sdx topology status"""
+        # pylint: disable=W0201
+        content = get_json_or_400(request, self.controller.loop)
+        self.sdx_topology = content.get("sdx_topology")
+        if self.sdx_topology is None:
+            self.sdx_topology = topology_mock.topology_mock()
+        response = self.validate_sdx_topology()
+        result = response["result"]
+        status_code = response["status_code"]
+        return JSONResponse(result, status_code)
+
+    @rest("v1/convert_topology/{event_type}/{event_timestamp}")
+    def get_converted_topology(self, request: Request) -> JSONResponse:
+        """ REST to return the converted sdx topology"""
+        event_type = request.path_params["event_type"]
+        event_timestamp = request.path_params["event_timestamp"]
+        response = self.convert_topology(event_type, event_timestamp)
+        result = response["result"]
+        status_code = response["status_code"]
+        return JSONResponse(result, status_code)
+
+    @rest("v1/post_sdx_topology/{event_type}/{event_timestamp}")
+    def get_sdx_topology(self, request: Request) -> JSONResponse:
+        """ REST to return the sdx topology loaded"""
+        event_type = request.path_params["event_type"]
+        event_timestamp = request.path_params["event_timestamp"]
+        response = self.post_sdx_topology(event_type, event_timestamp)
+        result = response["result"]
+        status_code = response["status_code"]
+        return JSONResponse(result, status_code)
+
+    @rest("v1/listen_event", methods=["POST"])
+    def get_listen_event(self, request: Request) -> JSONResponse:
+        """consume call listen Event"""
+        try:
+            result = get_json_or_400(request, self.controller.loop)
+            name = result.get("name")
+            content = result.get("content")
+            event = KytosEvent(
+                    name=name, content=content)
+            # self.controller.buffers.app.put(event)
+            event_result = self.listen_event(event)
```

Comment:
A decorated `@listen_to` method isn't meant to be called since its future is consumed internally by its thread pool, I'll recommend you reassess how you've structured this part, maybe you can extract a new method or func if you need.